src/middlewared/middlewared/plugins/disk_/disk_events.py

```diff
@@ -1,66 +1,48 @@
 import asyncio
-import re
 
-from middlewared.utils import osc
-
-
-if osc.IS_FREEBSD:
-    import sysctl
-    RE_ISDISK = re.compile(r'^(da|ada|vtbd|mfid|nvd|pmem)[0-9]+$')
+import sysctl
 
 
 async def added_disk(middleware, disk_name):
+    await middleware.call('geom.add_disk', disk_name)
```

Comment:
As `add_disk` is in reality invalidating the cache, probably better we don't hide that and rename this to reflect what's really happening